File: typescript/base-hmi-types/src/screens/shapes/HmiCircularArc.ts
import { HmiProperty, staticProperty } from "../base.js";
import { HmiCircularShapeBase } from "./HmiCircularShapeBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiCircularArc extends HmiCircularShapeBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiCircularArc;
  }

  startAngle: HmiProperty<number> = staticProperty(0);
  sweepAngle: HmiProperty<number> = staticProperty(0);
}

File: src/BaseHmiTypes/Screens/Shapes/HmiCircularArc.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Shapes;

public class HmiCircularArc : HmiCircularShapeBase
{
    public HmiCircularArc()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiCircularArc;
    }

    public HmiProperty<double> StartAngle { get; set; } = 0;

    public HmiProperty<double> SweepAngle { get; set; } = 0;
}
